roles/tidb_lightning/tasks/binary_deployment.yml

```diff
@@ -0,0 +1,32 @@
+---
+
+- name: deploy tidb-lightning binary
+  copy: src="{{ resources_dir }}/bin/tidb-lightning" dest="{{ deploy_dir }}/bin/" mode=0755 backup=yes
+  register: tidb_lightning
+
+- name: deploy tidb-lightning-ctl binary
+  copy: src="{{ resources_dir }}/bin/tidb-lightning-ctl" dest="{{ deploy_dir }}/bin/" mode=0755 backup=yes
+  register: tidb_lightning_ctl
+
+- name: backup tidb-lightning bianry file
+  command: mv "{{ tidb_lightning.backup_file }}" "{{ backup_dir }}"
+  when: tidb_lightning.changed and tidb_lightning.backup_file is defined
+
+- name: backup tidb-lightning-ctl bianry file
+  command: mv "{{ tidb_lightning_ctl.backup_file }}" "{{ backup_dir }}"
+  when: tidb_lightning_ctl.changed and tidb_lightning_ctl.backup_file is defined
+
+- name: create run script
+  template:
+    src: "{{ item }}_lightning_binary.sh.j2"
+    dest: "{{ deploy_dir }}/scripts/{{ item }}_lightning.sh"
+    mode: "0755"
+    backup: yes
+  with_items:
+    - start
+    - stop
+  register: lightning_script
+
+- name: backup script file
```

Comment:
Maybe lightning_script.backup_file is always undefined, you can use debug module to test it.
refer to https://github.com/pingcap/tidb-ansible/blob/master/roles/prometheus/tasks/main.yml#L31